File: api-xmlsignature/csharp/src/model/xades/XAdESBaseElement.cs
using System.Xml;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature.model.xades
{

	using Element = XmlElement;
	using Context = tr.gov.tubitak.uekae.esya.api.xmlsignature.Context;
	using XMLSignatureException = tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException;
	using Constants = tr.gov.tubitak.uekae.esya.api.xmlsignature.core.Constants;
	using BaseElement = tr.gov.tubitak.uekae.esya.api.xmlsignature.model.BaseElement;

	/// <summary>
	/// @author ahmety
	/// date: Jun 22, 2009
	/// </summary>
	public abstract class XAdESBaseElement : tr.gov.tubitak.uekae.esya.api.xmlsignature.model.BaseElement
	{

		public XAdESBaseElement(Context aContext) : base(aContext)
		{
		}

	   /// <summary>
	   ///  Construct XADESBaseElement from existing </summary>
	   ///  <param name="aElement"> xml element </param>
	   ///  <param name="aContext"> according to context </param>
	   ///  <exception cref="XMLSignatureException"> when structure is invalid or can not be
	   ///      resolved appropriately </exception>
//JAVA TO C# CONVERTER WARNING: Method 'throws' clauses are not available in .NET:
//ORIGINAL LINE: public XAdESBaseElement(XmlElement aElement, tr.gov.tubitak.uekae.esya.api.xmlsignature.Context aContext) throws tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException
		public XAdESBaseElement(Element aElement, Context aContext) : base(aElement, aContext)
		{
		}

		// base element
		public override string Namespace
		{
			get
			{
				return Constants.NS_XADES_1_3_2;
			}
		}

	}

}

File: api-xmlsignature/src/tr/gov/tubitak/uekae/esya/api/xmlsignature/model/xades/XAdESBaseElement.java
package tr.gov.tubitak.uekae.esya.api.xmlsignature.model.xades;

import org.w3c.dom.Element;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.Context;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.core.Constants;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.model.BaseElement;

/**
 * @author ahmety
 * date: Jun 22, 2009
 */
public abstract class XAdESBaseElement extends BaseElement
{

    public XAdESBaseElement(Context aContext)
    {
        super(aContext);
    }

   /**
    *  Construct XADESBaseElement from existing
    *  @param aElement xml element
    *  @param aContext according to context
    *  @throws XMLSignatureException when structure is invalid or can not be
    *      resolved appropriately
    */
    public XAdESBaseElement(Element aElement, Context aContext)
            throws XMLSignatureException
    {
        super(aElement, aContext);
    }

    // base element
    public String getNamespace()
    {
        return Constants.NS_XADES_1_3_2;
    }

}
